### ISSUE TITLE
Rule E3690 - Aurora MySQL Default EngineVersion Validation Issue

### ISSUE THREAD

**Issue by ced455** [aws-cloudformation/cfn-lint#3794]
### CloudFormation Lint Version

1.18.1

### What operating system are you using?

alpine

### Describe the bug

The CloudFormation linter rule E3690 - Validate DB Cluster Engine and Engine Version does not allow setting the EngineVersion for aurora-mysql to 8.0, which is required to use the default engine version for Aurora MySQL version 3. This restriction prevents us from leveraging AWS RDS’s default upgrade path for Aurora MySQL, causing our stacks to drift from AWS’s managed versions over time.

``` JSON
{
  "type": "8.0 is not one of ['5.7.mysql_aurora.2.11.1', '5.7.mysql_aurora.2.11.2', ... , '8.0.mysql_aurora.3.07.1']",
  "message": "8.0 is not one of the allowed EngineVersion values at src/mycfD.yml:983:7"
}

```

### Expected behavior

Per [AWS CloudFormation documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbcluster.html#cfn-rds-dbcluster-engineversion), setting EngineVersion: 8.0 should allow RDS to manage the default version for Aurora MySQL version 3. This setup ensures that any minor version updates provided by AWS are automatically applied, reducing maintenance overhead and stack drift.

Additional Information: We rely on the EngineVersion being set to 8.0 to avoid specific version pinning and allow RDS to update stacks as needed, without requiring manual adjustments.

### Reproduction template

``` YAML
  AuroraDBCluster:
    Type: AWS::RDS::DBCluster
    Properties:
      BackupRetentionPeriod: 30
      DBClusterIdentifier: !Join [ '-', [ !Ref EnvironmentName, !Ref EnvironmentVersion, !Ref ApplicationCode, 'cluster' ] ]
      Engine: aurora-mysql
      EngineVersion: 8.0
```
